lib/spack/spack/util/windows_registry.py

```diff
@@ -0,0 +1,224 @@
+    HKEY_CURRENT_CONFIG = winreg.HKEY_CURRENT_CONFIG
+    HKEY_PERFORMANCE_DATA = winreg.HKEY_PERFORMANCE_DATA
+
+
+class RegistryValue(object):
+    """
+    Class defining a Windows registry entry
+    """
+
+    def __init__(self, name, value, parent_key):
+        self.path = name
+        self.value = value
+        self.key = parent_key
+
+
+class RegistryKey(object):
+    """
+    Class wrapping a Windows registry key
+    """
+
+    def __init__(self, name, handle):
+        self.path = name
+        self.name = os.path.split(name)[-1]
+        self.handle = handle
+        self._keys = []
+        self._values = {}
+
+    @property
+    def values(self):
+        """Returns all subvalues of this key as RegistryValue objects in dictionary
+        of value name : RegistryValue object
+        """
+        self.gather_value_info()
+        return self._values
+
+    @property
+    def subkeys(self):
+        """Returns list of all subkeys of this key as RegistryKey objects"""
+        self.gather_subkey_info()
+        return self._keys
+
+    def gather_subkey_info(self):
+        """Composes all subjeys into a list for access"""
+        if self._keys:
+            return
+        sub_keys, _, _ = winreg.QueryInfoKey(self.handle)
+        for i in range(sub_keys):
+            sub_name = winreg.EnumKey(self.handle, i)
+            sub_handle = winreg.OpenKeyEx(self.handle, sub_name, access=winreg.KEY_READ)
+            self._keys.append(RegistryKey(os.path.join(self.name, sub_name), sub_handle))
+
+    def gather_value_info(self):
+        """Compose all values for this key into a dict of form value name: RegistryValue Object"""
+        if self._values:
+            return
+        _, values, _ = winreg.QueryInfoKey(self.handle)
+        for i in range(values):
+            value_name, value_data, _ = winreg.EnumValue(self.handle, i)
+            self._values[value_name] = RegistryValue(value_name, value_data, self.handle)
+
+    def get_subkey(self, sub_key):
+        """Returns subkey of name sub_key in a RegistryKey objects"""
+        return RegistryKey(
+            os.path.join(self.path, sub_key),
+            winreg.OpenKeyEx(self.handle, sub_key, access=winreg.KEY_READ),
+        )
+
+    def get_value(self, val_name):
+        """Returns value associated with this key in RegistryValue object"""
+        return RegistryValue(val_name, winreg.QueryValue(self.handle, val_name), self.handle)
+
+
+class WindowsRegistry(object):
+    """
+    Interface to provide access, querying, and searching to Windows registry entries.
+    This class represents a single key entrypoint into the Windows registry
+    and provides an interface to this key's values, its subkeys, and those subkey's values.
+    This class cannot be used to move freely about the registry, only subkeys/values of
+    the root key used to instantiate this class.
+    """
+
+    def __init__(self, key, root_string=winreg.HKEY_CLASSES_ROOT):
```

Comment:
This default value is forcing a search for `winreg` and I think leading to https://readthedocs.org/projects/spack/builds/18633158/